Make import notifications fail when any imports fail

peppy opened the pull request.

peppy committed:
Make import notifications fail when any imports fail

Aergwyn commented on osu.Game/Database/ArchiveModelManager.cs:
```diff
@@ -97,11 +98,11 @@ public void Import(params string[] paths)
                 try
                 {
-                    notification.Text = $"Importing ({i} of {paths.Length})\n{Path.GetFileName(path)}";
+                    notification.Text = $"Importing ({success} of {paths.Length})\n{Path.GetFileName(path)}";
```
While here do we want to address the "Importing 0 of 123" for readability?

peppy replied:
yeah i'll change that too

peppy committed:
Further improvements to messaging

Aergwyn commented on osu.Game/Database/ArchiveModelManager.cs:
```diff
@@ -331,7 +333,7 @@ private ArchiveReader getReaderFrom(string path)
             if (ZipFile.IsZipFile(path))
                 return new ZipArchiveReader(Files.Storage.GetStream(path), Path.GetFileName(path));
-            return new LegacyFilesystemReader(path);
+                return new LegacyFilesystemReader(path);
```
Nitpick: Sneaky whitespace

peppy committed:
Fix whitespace

smoogipoo commented on osu.Game/Database/ArchiveModelManager.cs:
```diff
@@ -97,11 +97,11 @@ public void Import(params string[] paths)
                         imported.Add(Import(reader));
 
-                    notification.Progress = (float)++i / paths.Length;
+                    notification.Progress = (float)(current - 1) / paths.Length;
```
Why `current - 1`? This means after importing the first beatmap, the progress is going to be 0. And if there is only one beatmap to import, the progress is always going to be 0?

peppy replied:
current is incremented above, right?

smoogipoo replied:
Yeah but imagine if `paths.Length == 1`, then we'd get to this line with `current = 1`, so `progress` would be `0 / paths.Length` `= 0`. Whereas previously it would do `1 / paths.Length`.

smoogipoo replied:
What, skipping one of the ticks completely? Since when? I'd much rather advocate for having a debounce from completing notifications so that the progress bar fills up completely before firing up a second notification.

peppy replied:
Is this visible to you? I couldn't see it in testing.

peppy committed:
Apply same logic fixes to delete/restore all notifications

smoogipoo commented on osu.Game/Database/ArchiveModelManager.cs:
```diff
@@ -218,8 +220,8 @@ public void Delete(List<TModel> items)
-                    notification.Text = $"Deleting ({i} of {items.Count})";
-                    notification.Progress = (float)++i / items.Count;
+                    notification.Progress = (float)i / items.Count;
+                    notification.Text = $"Deleting ({++i} of {items.Count})";
```
Almost the same thing as mentioned above, progress would be 0 for the first map. How about the following sequence instead:

```
notification.Text = "Deleting {++i} of ...";
Delete(b);
notification.Progress = (float)i / items.Count;
```

peppy replied:
is this even visible?

smoogipoo committed:
Update progress with the current item, not the next item

In the case where there is no next item, the progress will not get updated, so we'll essentially skip one element from filling the progress bar further. In the future we may/will want to not hide the notification upon completion, so this will look better in such scenarios.

peppy committed:
Merge branch 'master' into better-import-notice
